File: product-webapp/webapp/src/app/model/log.ts
export enum LogType {
  PROGRAM = 'PROGRAM',
  PROJECT = 'PROJECT',
  SELF_LEARNING = 'SELF_LEARNING',
}

export class Log {
  logType: LogType;
  logHours: number;
  taskName: string; // name of project or program
  startDate: number; // date on which program or project was started
  selfStudyDetails: string;

  constructor(
    logType: LogType,
    logHours: number,
    taskName: string,
    startDate:number,
    selfStudyDetails:string
  ) {
    this.logType = logType;
    this.logHours = logHours;
    this.taskName = taskName;
    this.startDate = startDate;
    this.selfStudyDetails = selfStudyDetails;
  }
}

File: time-tracker-service/src/main/java/com/stackroute/timetrackerservice/model/Log.java
package com.stackroute.timetrackerservice.model;

import lombok.AllArgsConstructor;
import lombok.Data;
import lombok.NoArgsConstructor;

@Data
@AllArgsConstructor
@NoArgsConstructor
public class Log {

    private String taskName;
    private LogType logType;
    private int logHours;
    private String startDate;
    private String selfStudyDetails;
}
